packages/sdk/src/oracle/errors.ts

```diff
@@ -0,0 +1,26 @@
+import { exists } from "../utils";
+
+// This error should be thrown if an expected value does not exist
+export class ExistenceError extends Error {
+  constructor(message = "") {
+    super(message);
+    this.name = "ExistenceError";
+    // if this isnt included, we cannot use instanceof to check the type
+    Object.setPrototypeOf(this, ExistenceError.prototype);
+  }
+}
+
+// Special assert which checks for existence and throw existence error
+export function assertExists<T>(condition: T, message = ""): asserts condition is NonNullable<T> {
```

Comment:
This return type syntax is cool!

Reply:
yea i had to do some research to figure this out!